esmf_regrid/tests/unit/esmf_regridder/test_Regridder.py

```diff
@@ -0,0 +1,116 @@
+"""Unit tests for :class:`esmf_regrid.esmf_regridder.Regridder`."""
+
+import numpy as np
+from numpy import ma
+import scipy.sparse
+
+from esmf_regrid.esmf_regridder import GridInfo, Regridder
+from esmf_regrid.tests import make_grid_args
+
+
+def _expected_weights():
+    weight_list = np.array(
+        [
+            0.6674194025656819,
+            0.3325805974343169,
+            0.3351257294386341,
+            0.6648742705613656,
+            0.33363933739884066,
+            0.1663606626011589,
+            0.333639337398841,
+            0.1663606626011591,
+            0.16742273275056854,
+            0.33250863479149745,
+            0.16742273275056876,
+            0.33250863479149767,
+            0.6674194025656823,
+            0.3325805974343174,
+            0.3351257294386344,
+            0.6648742705613663,
+        ]
+    )
+    rows = np.array([0, 0, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 5, 5])
+    columns = np.array([0, 1, 1, 2, 0, 1, 3, 4, 1, 2, 4, 5, 3, 4, 4, 5])
+
+    shape = (6, 6)
+
+    weights = scipy.sparse.csr_matrix((weight_list, (rows, columns)), shape=shape)
+    return weights
+
+
+def test_Regridder_init():
+    """Basic test for :meth:`~esmf_regrid.esmf_regridder.Regridder.__init__`."""
+    lon, lat, lon_bounds, lat_bounds = make_grid_args(2, 3)
+    src_grid = GridInfo(lon, lat, lon_bounds, lat_bounds)
+
+    lon, lat, lon_bounds, lat_bounds = make_grid_args(3, 2)
+    tgt_grid = GridInfo(lon, lat, lon_bounds, lat_bounds)
+
+    rg = Regridder(src_grid, tgt_grid)
+
+    result = rg.weight_matrix
+    expected = _expected_weights()
+
+    assert np.allclose(result.toarray(), expected.toarray())
+
+
+def test_Regridder_regrid():
+    """Basic test for :meth:`~esmf_regrid.esmf_regridder.Regridder.regrid`."""
+    lon, lat, lon_bounds, lat_bounds = make_grid_args(2, 3)
+    src_grid = GridInfo(lon, lat, lon_bounds, lat_bounds)
+
+    lon, lat, lon_bounds, lat_bounds = make_grid_args(3, 2)
+    tgt_grid = GridInfo(lon, lat, lon_bounds, lat_bounds)
+
+    # Set up the regridder with precomputed weights.
+    rg = Regridder(src_grid, tgt_grid, precomputed_weights=_expected_weights())
+
+    src_array = np.array([[1, 1, 1], [1, 0, 0]])
```

Comment:
This gives us integer arrays, which raises the question of dtype. Should the source dtype be the result dtype? Intermediate calculations always with double, even if source and target are single precision? Perhaps it's best to avoid those questions for now by using a floating point type here :smile:. As a rule of thumb, I find it useful to always use 1. etc where not explicitly integers are desired.